core/src/main/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequest.java

```diff
@@ -104,7 +104,7 @@ public ActionRequestValidationException validate() {
             validationException = addValidationError("name is missing", validationException);
         }
         if (indexPatterns == null || indexPatterns.size() == 0) {
-            validationException = addValidationError("pattern is missing", validationException);
+            validationException = addValidationError("index_patterns is missing", validationException);
```

Comment:
I wonder if this should be "index patterns are missing" as these requests can also come in from the transport layer where there is no notion of a "index_patterns" field?

Reply:
Makes sense. I pushed another change https://github.com/elastic/elasticsearch/pull/27102/commits/f81b8eae94d69b6fd6f748f0782649932f488e58.